__tests__/bin.js

```diff
@@ -0,0 +1,43 @@
+const { exec } = require("child_process");
+const { unlink } = require("fs");
+
+const test = require("ava").cb;
+
+const nodeVersion = `node: ${process.version.replace('v', '')}`;
+
+test("run with node", t => {
+  exec("node ./bin.js --print", (error, stdout) => {
```

Comment:
Would it be worth adding a test that runs each of `require('package.json').bin` as an executable directly? Such as: `exec("./bin.js")` -- I believe this represents how check-node-version was being run in the bug report.

Reply:
🤔 Yes, we could have the test determine the script to run from `bin`. I haven't thought of that at the time.

Running the script directly, however, does not work on Windows since the shebang is a *nix feature. `npm`/`npx` mimicks that by creating platform-specific intermediate scripts that call node and point it at the script when they run. In the bug report, the script was executed through `npx`, and that is covered by the final test which actually installs `check-node-version` inside itself and executes the generated intermediate script. That is exactly what happens on `npx check-node-version@4.0.1`, except that instead of downloading the package, the local one is used.

While cross-testing `index.js` is doable, because it relies on mockable features within the node environment, the same cannot be done here. The most we could do is add a conditional test that only runs on non-Windows platforms.

Reply:
Thanks for the explanation! I think what you have now is sufficient.